src/aliceVision/lensCorrectionProfile/lcp.cpp

```diff
@@ -0,0 +1,929 @@
+    _isSeqOpened(false),
+    _isCommonOK(false),
+    _isCamDataOK(false),
+    _inAlternateLensIDs(false),
+    _inAlternateLensNames(false),
+    _waitPerspModeldescription(false),
+    _getText(false),
+    _modelCount(0),
+    Author(""),
+    Make(""),
+    Model(""),
+    UniqueCameraModel(""),
+    CameraRawProfile(false),
+    LensInfo(""),
+    CameraPrettyName(""),
+    LensPrettyName(""),
+    ProfileName(""),
+    SensorFormatFactor(1.f),
+    ImageWidth(0),
+    ImageLength(0)
+{
+    load(filename, fullParsing);
+}
+
+void LCPinfo::load(const std::string& filename, bool fullParsing)
+{
+    XML_Parser parser = XML_ParserCreate(nullptr);
+
+    if (!parser) {
+        throw std::runtime_error("Couldn't allocate memory for XML parser");
+    }
+
+    if (fullParsing)
+    {
+        XML_SetElementHandler(parser, XmlStartHandler, XmlEndHandler);
+    }
+    else
+    {
+        XML_SetElementHandler(parser, XmlStartHandlerCommonOnly, XmlEndHandlerCommonOnly);
+    }
+    
+    XML_SetCharacterDataHandler(parser, XmlTextHandler);
+    XML_SetUserData(parser, static_cast<void*>(this));
+
+    FILE* const pFile = fopen(filename.c_str(), "rb");
+
+    if (pFile) {
+        constexpr int BufferSize = 8192;
+        char buf[BufferSize];
+        bool done;
+
+        do {
+            int bytesRead = fread(buf, 1, BufferSize, pFile);
+            done = feof(pFile);
+
+            if (XML_Parse(parser, buf, bytesRead, done) == XML_STATUS_ERROR) {
+                XML_ParserFree(parser);
+                throw std::runtime_error("Invalid XML in LCP file");
+            }
+        } while (!done);
+
+        fclose(pFile);
+    }
+
+    XML_ParserFree(parser);
+}
+
+bool LCPinfo::search(settingsInfo& settings, LCPCorrectionMode mode, int& iLow, int& iHigh, float& weightLow)
+{
+    iLow = iHigh = -1;
+
+    std::vector<bool> v_isDistortionValid;
+    std::vector<bool> v_isVignetteValid;
+
+    // Search best focal length
+    for (size_t i = 0; i < v_lensParams.size(); ++i)
+    {
+        const LensParam& currParam = v_lensParams[i];
+        const float f = currParam.camData.FocalLength;
+
+        v_isDistortionValid.push_back(currParam.isFisheye() ? !currParam.fisheyeParams.isEmpty : !currParam.perspParams.isEmpty);
+        v_isVignetteValid.push_back(currParam.hasVignetteParams() && !currParam.vignParams.isEmpty);
+
+        bool isCurrentValid = (mode == LCPCorrectionMode::DISTORTION && v_isDistortionValid.back()) ||
+                              (mode == LCPCorrectionMode::VIGNETTE && v_isVignetteValid.back());
+
+        if (isCurrentValid)
+        {
+            if (
+                f <= settings.FocalLength
+                && (
+                    iLow == -1
+                    || f > v_lensParams[iLow].camData.FocalLength
+                    || (settings.FocusDistance == 0 && f == v_lensParams[iLow].camData.FocalLength && v_lensParams[iLow].camData.FocusDistance > currParam.camData.FocusDistance)
```

Comment:
I think it makes sense to extract this into a function, especially since the code almost duplicated below. It could probably be a local lambda.